python/cugraph/cugraph/gnn/data_loading/dist_sampler.py

```diff
@@ -0,0 +1,349 @@
+                if has_weights
+                else cupy.array([], dtype="float32")
+            )
+
+            # create the renumber map offsets
+            renumber_map_offsets_array_p = minibatch_dict["renumber_map_offsets"][
+                partition_start : partition_end + 1
+            ]
+
+            renumber_map_start_ix, renumber_map_end_ix = renumber_map_offsets_array_p[
+                [0, -1]
+            ]
+
+            renumber_map_array_p = minibatch_dict["renumber_map"][
+                renumber_map_start_ix:renumber_map_end_ix
+            ]
+
+            results_dataframe_p = create_df_from_disjoint_arrays(
+                {
+                    "majors": majors_array_p,
+                    "minors": minors_array_p,
+                    "map": renumber_map_array_p,
+                    "label_hop_offsets": label_hop_offsets_array_p,
+                    "weight": weight_array_p,
+                    "edge_id": edge_id_array_p,
+                    "edge_type": edge_type_array_p,
+                    "renumber_map_offsets": renumber_map_offsets_array_p,
+                }
+            )
+
+            end_batch_id = start_batch_id + len(batch_id_array_p) - 1
+            rank = minibatch_dict["rank"] if "rank" in minibatch_dict else 0
+
+            full_output_path = os.path.join(
+                self.__directory,
+                f"batch={rank:05d}.{start_batch_id:08d}-"
+                f"{rank:05d}.{end_batch_id:08d}.parquet",
+            )
+
+            results_dataframe_p.to_parquet(
+                full_output_path,
+                compression=None,
+                index=False,
+                force_nullable_schema=True,
+            )
+
+    def __write_minibatches_csr(self, minibatch_dict):
+        raise NotImplementedError(
+            "CSR format currently not supported for distributed sampling"
+        )
+
+    def write_minibatches(self, minibatch_dict):
+        if (minibatch_dict["majors"] is not None) and (
+            minibatch_dict["minors"] is not None
+        ):
+            self.__write_minibatches_coo(minibatch_dict)
+        elif (minibatch_dict["major_offsets"] is not None) and (
+            minibatch_dict["minors"] is not None
+        ):
+            self.__write_minibatches_csr(minibatch_dict)
+        else:
+            raise ValueError("invalid columns")
+
+
+class DistSampler:
+    def __init__(
+        self,
+        graph: Union[pylibcugraph.SGGraph, pylibcugraph.MGGraph],
+        writer: DistSampleWriter,
+        local_seeds_per_call: int = 32768,
+    ):
+        self.__graph = graph
+        self.__writer = writer
+        self.__local_seeds_per_call = local_seeds_per_call
+
+    def sample_batches(
+        self, seeds: TensorType, batch_ids: TensorType, random_state: int = 0
+    ):
+        raise NotImplementedError("Must be implemented by subclass")
+
+    def sample_from_nodes(
+        self, nodes: TensorType, *, batch_size: int = 16, random_state: int = 62
```

Comment:
Can we a docstring here.  Hard to understand the inputs of this function. Are nodes all the seed we need to sample, if yes wont the below lines at 216 be very slow ? 

```
for i, current_seeds in enumerate(nodes):
```

Reply:
Nodes is a list of tensors.  Except it's actually a collection of zero-copy views of the original tensor created using `torch.split`.  Each tensor in nodes is the input for one sampling call.  So this isn't any slower than the bulk sampler was.  In fact, I think this approach is slightly more efficient.

Reply:
Can we add a docstring to reflect that please and update the type hint accordingly.

Reply:
Also changed the variable name instead of overwriting